packages/connect/src/api/firmware/x509certificate.ts

```diff
@@ -286,6 +286,64 @@ const parseValidity = (asn1: Asn1) => {
     };
 };
 
+/*
+ * https://www.alvestrand.no/objectid/2.5.29.html
+ * Extensions ::= SEQUENCE {
+ *   extnID      OBJECT IDENTIFIER,
+ *   critical    BOOLEAN OPTIONAL,
+ *   extnValue   OCTET STRING
+ * }
+ */
+const parseExtensions = (data: Asn1) => {
+    const asn1 = derToAsn1(data.contents);
+    if (asn1.cls !== 0 || asn1.tag !== 16 || !asn1.structured) {
+        throw new Error("This can't be a Extension. Wrong data type.");
+    }
+
+    const readBoolean = (value?: Uint8Array) => value && value[0] === 0xff;
```

Comment:
I would be mroe careful and check that `value[0]` is either `0xff` or `0x00`. This prevents a BER-encoded critical extension from being interpreted as non-critical.

Reply:
[6efe641](https://github.com/trezor/trezor-suite/pull/13173/commits/6efe641c4be87330e2c50345bc8c337f0db2fe26)